homeassistant/components/lg_thinq/climate.py

```diff
@@ -0,0 +1,344 @@
+                ClimateEntityFeature.TARGET_TEMPERATURE_RANGE
+            )
+
+    def _update_status(self) -> None:
+        """Update status itself."""
+        super()._update_status()
+
+        # Update fan, hvac and preset mode.
+        if self.data.is_on:
+            if self.supported_features & ClimateEntityFeature.FAN_MODE:
+                self._attr_fan_mode = self.data.fan_mode
+
+            hvac_mode = self._requested_hvac_mode or self.data.hvac_mode
+            if hvac_mode in STR_TO_HVAC:
+                self._attr_hvac_mode = STR_TO_HVAC.get(hvac_mode)
+                self._attr_preset_mode = None
+            elif hvac_mode in THINQ_PRESET_MODE:
+                self._attr_preset_mode = hvac_mode
+        else:
+            if self.supported_features & ClimateEntityFeature.FAN_MODE:
+                self._attr_fan_mode = FAN_OFF
+
+            self._attr_hvac_mode = HVACMode.OFF
+            self._attr_preset_mode = None
+
+        self.reset_requested_hvac_mode()
+        self._attr_current_humidity = self.data.humidity
+        self._attr_current_temperature = self.data.current_temp
+        self._attr_temperature_unit = (
+            self._get_unit_of_measurement(self.data.unit) or UnitOfTemperature.CELSIUS
+        )
+
+        if (max_temp := self.entity_description.max_temp or self.data.max) is not None:
+            self._attr_max_temp = max_temp
+
+        if (min_temp := self.entity_description.min_temp or self.data.min) is not None:
+            self._attr_min_temp = min_temp
+
+        if (step := self.entity_description.step or self.data.step) is not None:
+            self._attr_target_temperature_step = self._correct_step(step)
+
+        # Update target temperatures.
+        if (
+            self.supported_features & ClimateEntityFeature.TARGET_TEMPERATURE_RANGE
+            and self.hvac_mode == HVACMode.AUTO
+        ):
+            self._attr_target_temperature = None
+            self._attr_target_temperature_high = self.data.target_temp_high
+            self._attr_target_temperature_low = self.data.target_temp_low
+        else:
+            self._attr_target_temperature = self.data.target_temp
+            self._attr_target_temperature_high = None
+            self._attr_target_temperature_low = None
+
+        _LOGGER.debug(
+            "[%s:%s] update status: %s/%s -> %s/%s, hvac:%s, unit:%s",
+            self.coordinator.device_name,
+            self.property_id,
+            self.data.current_temp,
+            self.data.target_temp,
+            self.current_temperature,
+            self.target_temperature,
+            self.hvac_mode,
+            self.temperature_unit,
+        )
+
+    def _correct_step(self, step: float) -> float:
+        """Correct step if needed."""
+        hass_unit = self.coordinator.hass.config.units.temperature_unit
+        if self.temperature_unit != hass_unit and step < 1:
+            return 1
+
+        return step
```

Comment:
Why is the step affected by the temperature unit?

Reply:
For example,
when step is 0.5 and unit is F, 
it causes the calculation error.
Keep it as is.

Reply:
Causes calculation error where?

Reply:
When device's step is 0.5 and HA's config's unit is Fahrenheit, then async_set_temperature has same output value
by +, - climate's button.

For example 65.0 and 65.5 has same output converting to Celsius by _round_by_step.
1. 65.0F -> 18.333333333333332 -> 18.5
2. 65.5F -> 18.61111111111111 -> 18.5

So, if it is less than 1, it needs to be corrected to 1.